pulpcore/cli/file/repository.py

```diff
@@ -60,6 +70,9 @@ def repository(ctx: click.Context, pulp_ctx: PulpContext, repo_type: str) -> Non
 
 
 lookup_options = [href_option, name_option]
+nested_lookup_options = [
+    click.option("--repository", callback=_repository_callback, expose_value=False)
+]
```

Comment:
I am wondering, if this kind of lookup should be used for the version and label command groups too.

Reply:
It might help to see some code. At least for labels, I can't visualize a benefit.

Reply:
The problem is, that we cannot use `--name` and `--href` here because the `TaskContext` would ingest them.
So we are left with `--repository` and maybe `--repository-href`. For `label` it would be for the sake of consistency.